src/sdam/topology.ts

```diff
@@ -290,8 +293,13 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
     const topologyType = topologyTypeFromOptions(options);
     const topologyId = globalTopologyCounter++;
 
+    const selectedHosts =
+      options.srvMaxHosts === 0 || options.srvMaxHosts >= seedlist.length
+        ? seedlist
+        : shuffle(seedlist, options.srvMaxHosts);
```

Comment:
I moved the initial shuffling to here, in order to unit test since the callback to resolveSRVRecord is inaccessible to the tests.

Reply:
seems like that works too